internal/provider/data_source_consul_cluster.go

```diff
@@ -300,6 +319,21 @@ func setConsulClusterDataSourceAttributes(
 		}
 	}
 
+	if cluster.Config.NetworkConfig != nil && cluster.Config.NetworkConfig.IPAllowlist != nil {
```

Comment:
Not sure if I need this `nil` check, maybe it's too defensive. Assuming the NetworkConfig will always exist when reading from a cluster